intellij/src/saros/intellij/context/SarosIntellijContextFactory.java

```diff
@@ -73,8 +69,6 @@
       Component.create(UIProjectUtils.class),
 
       // IDE-specific classes for the HTML GUI
-      Component.create(DialogManager.class, IntellijDialogManager.class),
-      Component.create(IUIResourceLocator.class, IntelliJUIResourceLocator.class),
       Component.create(ICollaborationUtils.class, IntellijCollaborationUtilsImpl.class),
```

Comment:
I was not sure whether the ICollaborationUtils is a nice approach of creating an abstraction layer that can be used by the GUI.... What do you think?

Reply:
Both implementations contains UI code (dialog popups). The Eclipse implementation does some prechecks  and some partial stuff. If you want to extract some code into the core feel free to do it.

Reply:
Well `IWorkspaceRoot` is currently not implemented as the concept of a 'workspace root' does not fit into the IntelliJ module concept. I think it was introduced to have a base relative to which to display the shared resources in the HTML UI. Now that this is not an issue anymore, I don't see a need for this class anymore. So I am currently against keeping it around. But I am open to discussion if you have a different option.

As for `ICollaborationsUtils`, I just mentioned it as it is no longer actively called anywhere.
I am not sure whether it is actually needed as an abstraction. It seems to me as if the complete functionality of the interface were already provided by [`ISarosSessionManager`](https://github.com/saros-project/saros/blob/master/core/src/saros/session/ISarosSessionManager.java). The only part that is contained in the implementation (and not even directly for Saros/I, as it is contained in [`CollaborationUtils`](https://github.com/saros-project/saros/blob/master/intellij/src/saros/core/ui/util/CollaborationUtils.java), which does not implement `ICollaborationUtils` directly) is the combination of some calls to `ISarosSessionManager` and the user notification stuff (progress bar, dialogs, etc.).
This could also be moved to the core, but it would require the dialog utils also being available in the core. This is difficult for Saros/I as dialogs are dependent on the shared project (or rather the project to bind the dialog to), meaning it is hard to find an IDE-independent way to call them.

Reply:
I will check ISarosSessionManager and the ICollaborationUtils and make the changes in a follow-up PR or discuss my thoughts in a separate issue.